MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -1083,27 +1083,50 @@ public bool CanStreamCopyAudio(EncodingJobInfo state, MediaStream audioStream, s
         {
             var bitrate = request.VideoBitRate;
 
-            // If specific values were requested, then force the caller to supply a bitrate as well
-            if (request.Height.HasValue && request.Width.HasValue)
-            {
-                return bitrate;
-            }
-
             if (videoStream != null)
             {
-                if (bitrate.HasValue)
-                {
-                    var inputVideoCodec = videoStream.Codec;
-                    bitrate = ScaleBitrate(bitrate.Value, inputVideoCodec, outputVideoCodec);
+                var isUpscaling = request.Height.HasValue && videoStream.Height.HasValue &&
+                                   request.Height.Value > videoStream.Height.Value && request.Width.HasValue && videoStream.Width.HasValue &&
+                    request.Width.Value > videoStream.Width.Value;
 
-                    // If a max bitrate was requested, don't let the scaled bitrate exceed it
-                    if (request.VideoBitRate.HasValue)
+                // Don't allow bitrate increases unless upscaling
```

Comment:
What was the thought behind this again?

Reply:
Well, in the previous state, the bitrate was minimized in every case. Now, it is only minimized when it is not upscaled; when upscaling, higher bitrate values could be needed.

Reply:
All this commit is basically a port of how the recent versions of Emby are handling bitrate for transcoding. I have been using it for months, with very good results, that is, a well-tuned balance between keeping a good video quality and not overloading the bandwidth.

Reply:
It looks like there are a lot of magic numbers in here that I really would prefer to steer clear of (why 2.5x when under 2mbps and 2x when under 3mbps, but no increase when above those two amounts?). I'm guessing the increase is so that transcoding can use less-efficient codecs if the client doesn't support the more efficient original codec?

Reply:
About the differences in the efficiency of codecs, I guess you already have noticed that [there is a function that already takes  that into consideration](https://github.com/jellyfin/jellyfin/blob/012e4a3e63f8b0997d5f070b37ac93d257f894e7/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L1133).

About those "magic numbers" you mention, as I said, that was introduced in latest versions of Emby, I assume after lots of trial and error with different materials. The reason why I included them there is just because they happen to work very well  :)  I have several users accessing my media server from the Internet, and I can tell you that this combination makes a very good balance between keeping a good video quality, and not creating bitrate spikes that would cause the video to stutter on limited bandwith situations.

Of course, they can be further tuned up (everything is subject to improvement, right?), but I think they are a good starting point.

Reply:
This should have a config variable, since really the only limitation for LAN streaming is the interface speed, so for wired clients this could just as easily be the 140Mbps (configurable) standard it used to be. Especially since that will allow for better quality and faster encodes since you can use faster presets without huge quality loss.

I don't agree that a lossy to lossy transcode should have the same bitrate at both ends. It should probably be much more on the output end to reduce quality losses.

Reply:
> GetMinBitrate looks fine as a starting point if you include a comment that the chosen values are magic numbers and might benefit from more testing. I agree that video streams with low bitrates are noticably worse than I would expect so any attempt to improve that is welcome.

@dkanada So should I make another PR, adding the comment you mention?

> This should have a config variable
@EraYaN Definitely a good idea, to be implemented when someone can do it, since it involves quite a bit of work in designing a specific interface in the WebUI for this feature, etc.. In the meantime, I think this code improves the current situation.